python/python/lance/dataset.py

```diff
@@ -1059,12 +1059,15 @@ def create_index(
                             f"Ivf centroids must be 2D array: (clusters, dim), "
                             f"got {ivf_centroids.shape}"
                         )
-                    if ivf_centroids.dtype != np.float32:
+                    dim = ivf_centroids.shape[1]
+                    if ivf_centroids.dtype == np.float32:
+                        values = pa.array(ivf_centroids.reshape(-1), type=pa.float32())
+                    elif ivf_centroids.dtype == np.float16:
```

Comment:
what if they are float64 or bf16?

Reply:
let's handle those later I guess. np doesn't have bf16 I think. fp64 probably has too little use cases in ML

Reply:
Bfloat16 is in the ml types package from Jax. Without it, it seems to become a problem very soon ?